### ISSUE TITLE
AggregateColumn and how to use it

### ISSUE THREAD

**Issue by frefardt** [etlbox/etlbox#78]
Hi Andreas

I'm currently struggling with the Aggregation Transformation. I would like to set an AggregatedColumn like this within a Aggregation:

`AggregateColumn aggregateColumn = new AggregateColumn("Col2", AggregationMethod.Sum);`

But I can't find a way to do that. What am I missing?

Regards Florian

**Comment by andreaslennartz** [comment]
Hi @frefardt ,

thanks for the question. 
The AggregateColumn is actually an attribute which you place before properties.
So it can't be added to Aggregation transformation directly (yet).
Actually I will think about it if this is a use case I want to support - that you can pass aggregate columns to the Aggregation directly. 

Anyways, I updated the documentation for the Aggregation a lot, it now should explain much better how to use it: https://etlbox.net/articles/transformations/aggregating_data.html#example-aggregations-on-all-data

If you have feedback on this, please let me know - any feedback is appreciated.

**Comment by andreaslennartz** [comment]
Hi  @frefardt ,

thanks to your suggestion, I am currently working on adding two new Properties to the Aggregation: 

```
public IEnumerable<AggregateColumn> AggregateColumns { get; set; }
public IEnumerable<GroupColumn> GroupColumns { get; set; }
```
This will allow you to pass a list of AggregateColumn and GroupColumn to the Aggregation.This will work with ExpandoObjects as well. It is probably easier to use than working with the AggregationAction / GroupingFunc / StoreKeyAction. 

I'll keep you updated on the progress.

**Comment by frefardt** [comment]
Hi @andreaslennartz 

This would be super comfortable and much easier to use in deed. I'm looking forward to this 👍.

Regards Florian

**Comment by andreaslennartz** [comment]
Hi @frefardt ,

thanks for you feedback. I just released version 2.1.3: https://www.nuget.org/packages/ETLBox/2.1.3
This version allows you to pass AggregateColumn / GroupColumn as object to the AggregateColumns / GroupColumns property of the aggregate. 
I also updated the documentation how to use this: https://etlbox.net/articles/transformations/aggregating_data.html#define-aggregationcolumns-without-attributes.

Let me know if this works for you.
